tuist/tuist
# Deployment target derived correctly for tuist test

Resolves https://github.com/tuist/tuist/issues/2670

### Short description 📝

According to what @fortmarek said [here](https://github.com/tuist/tuist/issues/2670#issuecomment-1737991028), I've checked what's happening when `deploymentTarget` is set, or not:

When `deploymentTarget` parameter is set in the Project manifest, `project.pbxproj` has this value `IPHONEOS_DEPLOYMENT_TARGET = 17.0;` set explicitly.
When there is no `deploymentTarget` parameter in the Project manifest, `IPHONEOS_DEPLOYMENT_TARGET` does not exist in pbxproj, and Xcode selects the newest version.

Looking at this behavior, my proposal is to take the newest runtime version, and run tests on the first available simulator with that version

### How to test the changes locally 🧐

- use `ios_app_with_tests` fixture
- remove `deploymentTarget` property
- boot simulator with lower iOS version, than the newest one
- run `tuist test` on `ios_app_with_tests` fixture
- tests should run on simulator with the newest version

### Contributor checklist ✅


### Reviewer checklist ✅
